exchanges/okx/okx.go

```diff
@@ -0,0 +1,4737 @@
+	type response struct {
+		Msg  string                    `json:"msg"`
+		Code string                    `json:"code"`
+		Data []FundingTransferResponse `json:"data"`
+	}
+	var resp response
+	if arg == nil {
+		return nil, errors.New("argument can not be null")
+	}
+	if arg.Amount <= 0 {
+		return nil, errors.New("invalid funding amount")
+	}
+	if arg.Currency == "" {
+		return nil, errors.New("invalid currency value")
+	}
+	if !(strings.EqualFold(arg.From, "6") || strings.EqualFold(arg.From, "18")) {
+		return nil, errors.New("missing funding source field \"From\"")
+	}
+	if arg.To == "" {
+		return nil, errors.New("missing funding destination field \"To\"")
+	}
+	if arg.Type >= 0 && arg.Type <= 4 {
+		if arg.Type == 1 || arg.Type == 2 {
+			if arg.SubAccount == "" {
+				return nil, errors.New("subaccount name required")
+			}
+		}
+	} else {
+		return nil, errors.New("invalid reqest type")
+	}
+	return resp.Data, ok.SendHTTPRequest(ctx, exchange.RestSpot, fundsTransferEPL, http.MethodPost, assetTransfer, arg, &resp, true)
+}
+
+// GetFundsTransferState get funding rate response.
+func (ok *Okx) GetFundsTransferState(ctx context.Context, transferID, clientID string, transferType int) ([]TransferFundRateResponse, error) {
+	params := url.Values{}
+	switch {
+	case transferID == "" && clientID == "":
+		return nil, errors.New("either 'transfer id' or 'client id' is required")
+	case transferID != "":
+		params.Set("transId", transferID)
+	case clientID != "":
+		if !((Letters.MatchString(clientID) || Numbers.MatchString(clientID)) && clientID != "") {
+			return nil, errors.New("invalid client id")
+		}
+		params.Set("clientId", clientID)
+	}
+	if transferType > 0 && transferType <= 4 {
+		params.Set("type", strconv.Itoa(transferType))
+	}
+	path := common.EncodeURLValues(assetTransferState, params)
+	type response struct {
+		Code string                     `json:"code"`
+		Msg  string                     `json:"msg"`
+		Data []TransferFundRateResponse `json:"data"`
+	}
+	var resp response
+	return resp.Data, ok.SendHTTPRequest(ctx, exchange.RestSpot, getFundsTransferStateEPL, http.MethodGet, path, nil, &resp, true)
+}
+
+//  AssetBillsDeta Query the billing record, you can get the latest 1 month historical data
+func (ok *Okx) GetAssetBillsDetails(ctx context.Context, currency string, billType int, clientID, clientSecret string, after, before time.Time, limit int) ([]AssetBillDetail, error) {
+	params := url.Values{}
+	billTypeIDS := []int{1, 2, 13, 20, 21, 28, 41, 42, 47, 48, 49, 50, 51, 52, 53, 54, 59, 60, 61, 68, 69, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 91, 92, 93, 94, 95, 96, 97, 98, 99, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130, 131, 150, 151, 198, 199}
+	for x := range billTypeIDS {
+		if billTypeIDS[x] == billType {
+			params.Set("type", strconv.Itoa(billType))
+		}
+	}
```

Comment:
The bill type can be among the numeric types listed on the billTypeIDS string. So, I used a loop to iterate over a loop and check for each value.
But, I changed it to a single IF statement because I found some patterns in it.